### PR TITLE
Enable the build with the latest develop branches

### PR DESCRIPTION
**Pull request by liujake**
TYPE: maintenance

KEYWORDS: build, spack-stack

DESCRIPTION OF CHANGES: 
The build of the latest code needs to use the spack-stack build environment.

LIST OF MODIFIED FILES:
M       CMakeLists.txt
M       env-setup/gnu-openmpi-cheyenne.csh
M       env-setup/gnu-openmpi-cheyenne.sh
M       env-setup/intel-impi-cheyenne.csh
M       env-setup/intel-impi-cheyenne.sh

TESTS CONDUCTED: Build and ctest on Cheyenne.

### CODE REVIEW

**Comment by liujake** [comment]
I have the trouble (on Cheyenne) in the ecbuild step for CRTM with the error message:
```
-- Adding bundle project crtm

-- Cloning crtm from https://github.com/JCSDA/crtm.git into /glade/scratch/liuz/public_jcsda/code/crtm...

-- /glade/scratch/liuz/public_jcsda/code/crtm retrieved.

-- Updating crtm to head of BRANCH release/crtm_jedi...

-- git fetch --all @ /glade/scratch/liuz/public_jcsda/code/crtm

-- git fetch --all --tags @ /glade/scratch/liuz/public_jcsda/code/crtm

-- git checkout release/crtm_jedi @ /glade/scratch/liuz/public_jcsda/code/crtm

-- ---------------------------------------------------------

-- [crtm] (2.3.0) [bfede42]

-- Feature TESTS enabled

CMake Error at /glade/scratch/liuz/public_jcsda/build/jedicmake/jedicmake-config.cmake:19 (message):

  File or directory

  /glade/scratch/liuz/public_jcsda/code/jedicmake/CMakeModules/Modules

  referenced by variable jedicmake_MODULES does not exist !

Call Stack (most recent call first):

  /glade/scratch/liuz/public_jcsda/build/jedicmake/jedicmake-config.cmake:36 (set_and_check)

  crtm/CMakeLists.txt:20 (find_package)
```

**Comment by liujake** [comment]
@climbfuji pointed out that the problem was because the jedi-cmake line missing "RECURSIVE". Now it passed ecbuild step with RECURSIVE added. Building it now ....

**Comment by liujake** [comment]
Build and ctest succeeded on Cheyenne with gnu. Ready to review.
```
100% tests passed, 0 tests failed out of 39

Label Time Summary:
executable    =  19.61 sec*proc (7 tests)
mpasjedi      = 444.22 sec*proc (39 tests)
mpi           = 430.48 sec*proc (35 tests)
script        = 424.61 sec*proc (32 tests)

Total Test time (real) = 444.70 sec
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -21,27 +21,27 @@ find_package( ecbuild 3.5 REQUIRED HINTS ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CUR
 include( ecbuild_bundle )
 ecbuild_bundle_initialize()
 
-ecbuild_bundle( PROJECT jedicmake GIT "https://github.com/jcsda-internal/jedi-cmake.git" BRANCH develop UPDATE )
+ecbuild_bundle( PROJECT jedicmake GIT "https://github.com/JCSDA/jedi-cmake.git" BRANCH develop UPDATE RECURSIVE)
```
Comment (climbfuji) [inline CMakeLists.txt:24]: Note for the future. You can use jedi-cmake from spack-stack if it's available, or fall back to building it here. See: https://github.com/JCSDA/jedi-bundle/blob/e6037876c15e7cf0236cb6d41c83b59a18239ad8/CMakeLists.txt#L24

**Review by climbfuji — APPROVED** [review]

**Review by ibanos90 — APPROVED** [review]
Thanks for this PR @liujake!

**Review by ibanos90 — APPROVED** [review]

**Pull request merged**